### DISCUSSION TITLE
environment setup for tika/gotenberg in docker scenario

### DISCUSSION THREAD

**Discussion by kurti500** [jonaswinkler/paperless-ng#908] [Support]
Hi community,
for the environment variables in the docker compose file for gotenberg and tika, I am wondering what's the right/best setup for the endpoints: referencing localhost as shown below, entering the dedicated IP of the server, or going with name of the service as shown in the documentation https://paperless-ng.readthedocs.io/en/latest/configuration.html#tika-settings

    environment:
      PAPERLESS_REDIS: redis://broker:6379
      PAPERLESS_DBHOST: db
      PAPERLESS_TIKA_ENABLED: 1
      PAPERLESS_TIKA_GOTENBERG_ENDPOINT: http://localhost:3000
      PAPERLESS_TIKA_ENDPOINT: http://localhost:9998

I am getting sometimes errors like:

`[2021-04-13 13:02:01,497] [ERROR] [paperless.consumer] Error while consuming document ID(181746_2)_Date(Mon, 16 Jan 2017 11_53_04 +0100)_Bruno macantay - EN.DOC: Error while converting document to PDF: 400 Client Error: Bad Request for url: http://gotenberg:3000/convert/office

Traceback (most recent call last):

  File "/usr/src/paperless/src/paperless_tika/parsers.py", line 79, in convert_to_pdf

    response.raise_for_status()  # ensure we notice bad responses

  File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 943, in raise_for_status

    raise HTTPError(http_error_msg, response=self)

requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://gotenberg:3000/convert/office

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/usr/src/paperless/src/documents/consumer.py", line 248, in try_consume_file

    document_parser.parse(self.path, mime_type, self.filename)

  File "/usr/src/paperless/src/paperless_tika/parsers.py", line 65, in parse

    self.archive_path = self.convert_to_pdf(document_path, file_name)

  File "/usr/src/paperless/src/paperless_tika/parsers.py", line 82, in convert_to_pdf

    f"Error while converting document to PDF: {err}"

documents.parsers.ParseError: Error while converting document to PDF: 400 Client Error: Bad Request for url: http://gotenberg:3000/convert/office`

or

`[2021-04-13` 11:25:02,896] [ERROR] [paperless.consumer] Error while consuming document ID(255_3)_Date(Fri, 10 Aug 2001 20_03_10 +0200)_ABI93A.XLS: Could not parse /usr/src/paperless/src/../consume/ID(255_3)_Date(Fri, 10 Aug 2001 20_03_10 +0200)_ABI93A.XLS with tika server at http://192.168.1.60:9998: HTTPConnectionPool(host='192.168.1.60', port=9998): Max retries exceeded with url: /rmeta/text (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f9c7715ab10>: Failed to establish a new connection: [Errno 111] Connection `refused'))`

and wondering if this is related to how those services are setup?
Can anybody help? Any suggestions to avoid those errors?
Cheers,
Volker

**Comment by chessgriffin** [comment]
I am getting this same error.

**Comment by TobenderZephyr** [comment]
docker automatically resolves service names to its ip addresses internally, so you mustn't worry about correct IP adressing. That is, as long as you are using the same docker-compose file (because it creates a seperate unrouted network for each docker-compose you run.)
Else you would need to create a new network which acts as a backend network just between those services and specify those in all compose-files.

If you are not going to use gotenberg and tika somewhere else, it is easier just pasting the `services` into your current paperless compose file (you won't need to expose ports, too).

in short: just use service names and be sure to include both services in your paperlesses docker-compose.yml.

**Comment by bergfex** [comment]
I get the following error when I import an Open Office document:

`[2021-10-27 17:30:22,355] [INFO] [paperless.parsing.tika] Sending /tmp/paperless/paperless-upload-a529xvhw to Tika server
[2021-10-27 17:30:23,068] [INFO] [paperless.parsing.tika] Converting /tmp/paperless/paperless-upload-a529xvhw to PDF as /tmp/paperless/paperless-35axwwsk/convert.pdf
[2021-10-27 17:30:23,082] [DEBUG] [paperless.parsing.tika] Deleting directory /tmp/paperless/paperless-35axwwsk
[2021-10-27 17:30:23,088] [ERROR] [paperless.consumer] Error while consuming document Rechnung Okt.2021.odt: Error while converting document to PDF: 404 Client Error: Not Found for url: http://gotenberg:3000/convert/office
Traceback (most recent call last):
  File "/usr/src/paperless/src/paperless_tika/parsers.py", line 79, in convert_to_pdf
    response.raise_for_status()  # ensure we notice bad responses
  File "/usr/local/lib/python3.9/site-packages/requests/models.py", line 953, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://gotenberg:3000/convert/office
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/src/paperless/src/documents/consumer.py", line 248, in try_consume_file
    document_parser.parse(self.path, mime_type, self.filename)
  File "/usr/src/paperless/src/paperless_tika/parsers.py", line 65, in parse
    self.archive_path = self.convert_to_pdf(document_path, file_name)
  File "/usr/src/paperless/src/paperless_tika/parsers.py", line 81, in convert_to_pdf
    raise ParseError(
documents.parsers.ParseError: Error while converting document to PDF: 404 Client Error: Not Found for url: http://gotenberg:3000/convert/office`

The error has occurred since I updated to version 1.5.0. Before that, Open Office documents were imported and processed flawlessly. Now it aborts with the above error message. Everything else works fine. There is a small difference in the error message. For me it is a 404 error instead of 400.
`... raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://gotenberg:3000/convert/office`
`... documents.parsers.ParseError: Error while converting document to PDF: 404 Client Error: Not Found for url: http://gotenberg:3000/convert/office`

For the container I use the docker-compose file from the repository.

My environment:
- Debian 11
- Docker version 20.10.10
- docker-compose version 1.29.2
- Paperless-NG 1.5.0

I have already removed the container and the images and created them again.

How can I proceed further to isolate the error?


_Translated with Deepl.com_

**Comment by TobenderZephyr** [comment]
Since this topic is not entirely about paperless, but tika / gotenberg, maybe you try other configurations for this?

I would tag the docker images inside your docker-compose.yml with a fixed version number. Maybe the underlying framework or API calls have been changed and we don't know about it?

For Tika and gotenberg, i wouldn't want to use the latest versions, since I'm unaware of when support for both has been added/tested.
```
  tika:
    image: apache/tika:1.27

  gotenberg:
    image: thecodingmachine/gotenberg:6
```

But those are only hints, since I don't use any of those containers myself.

**Reply by bergfex** [reply]
Hello

Thank you for the answer and your advice.

Yesterday I installed Paperless-NG with Docker on my desktop computer. The same error pattern.

Then I will follow your hint.

**Reply by capi** [reply]
That solved it for me, thanks!

**Comment by leginreklaw** [comment]
I experienced the same issue with V1.5.0 and can confirm that specifying image V1.27 for Tika and V6 for Gotenberg solves the issue for me.
